features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Scopes/EditScope.jsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import React from 'react';
+import APIPropertyField from 'AppComponents/Apis/Details/Overview/APIPropertyField';
+import Typography from '@material-ui/core/Typography';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import TagsInput from 'react-tagsinput';
+import Api from 'AppData/api';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { Link } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import Alert from 'AppComponents/Shared/Alert';
+
+const styles = theme => ({
+    buttonSave: {
+        marginTop: theme.spacing.unit * 10,
+    },
+    buttonCancel: {
+        marginTop: theme.spacing.unit * 10,
+        marginLeft: theme.spacing.unit * 5,
+    },
+    topics: {
+        marginTop: theme.spacing.unit * 10,
+    },
+    headline: {
+        paddingTop: theme.spacing.unit * 1.5,
+        paddingLeft: theme.spacing.unit * 2.5,
+    },
+});
+
+/**
+ * Create new scopes for an API
+ * @class CreateScope
+ * @extends {Component}
+ */
+class EditScope extends React.Component {
+    constructor(props) {
+        super(props);
+        // this.api = new Api();
+        this.api_uuid = props.match.params.api_uuid;
+        const { api, location } = this.props;
+        this.state = {
+            apiScope: api.scopes.filter((scope) => {
+                return scope.name === location.state.scopeName;
+            })[0],
+        };
+        this.updateScope = this.updateScope.bind(this);
+        this.handleInputs = this.handleInputs.bind(this);
+    }
+
+    /**
+     * Add new scope
+     * @memberof Scopes
+     */
+    updateScope() {
+        const restApi = new Api();
+        const updatedScope = this.state.apiScope;
+        const { intl, api, history } = this.props;
+        api.scopes = api.scopes.map((scope) => {
```

Comment:
This is wrong, You are directly mutating the `props` value. Don't update the `api.scopes` directly.
```suggestion
const scopes = api.scopes.map((scope) => {
```